src/bin/generate.rs

```diff
@@ -16,13 +17,20 @@ Options:
   --email EMAIL      HoloPort admin email address
   --password STRING  HoloPort admin password
   --seed-from PATH   Use SHA-512 hash of given file truncated to 256 bits as seed
+  --encrypt          Encrypt the seed using the admin password
 ";
 
 #[derive(Deserialize)]
 struct Args {
     flag_email: String,
     flag_password: String,
     flag_seed_from: Option<PathBuf>,
+    flag_encrypt: bool,
+}
+
+pub fn to_url(hcid: &str) -> Result<Url, Error> {
```

Comment:
This should accept public key as an argument, rather than hcid. The fact that it uses hcid notation is incidental, and it even uses a different version of it (lowercase).